edgemesh/pkg/server/dns.go

```diff
@@ -93,10 +93,11 @@ func startDnsServer() {
 		Port: 53,
 	}
 	udpConn, err := net.ListenUDP("udp", laddr)
-	defer udpConn.Close()
 	if err != nil {
 		klog.Errorf("Dns server Start error : %s", err)
+		return
 	}
+	defer udpConn.Close()
```

Comment:
does this line cause a panic if it is above the if condition ?

Reply:
No, but the returned conn will be nil, it is no need to close it, like a nil file.
And the following process is also not needed, am I right?

Reply:
Yes, if it is above the if condition, it will cause a panic.
e.g. https://play.golang.org/p/2SpVzVrnGjg